test_api.py

```diff
@@ -1871,32 +1874,35 @@ def __iter__(self, *args, **kwargs):
             self._delete_hosts()
             return iterator
 
-    def test_delete_when_one_host_is_deleted(self, emit_event):
+    def test_delete_when_one_host_is_deleted(self):
         host_id = self.added_hosts[0].id
         url = HOST_URL + "/" + host_id
         with patch("api.host.delete_hosts", self.DeleteHostsMock.create_mock([host_id])):
             # One host queried, but deleted by a different process. No event emitted yet returning
             # 200 OK.
-            self.delete(url, 200, return_response_as_json=False)
-            emit_event.assert_not_called()
+            with self.app.app_context():
+                self.delete(url, 200, return_response_as_json=False)
+                self.event_producer.write_event.assert_not_called()
```

Comment:
It’s _self.app.event_producer_ that is used by _delete_hosts_. This assertion would thus pass, even when it shouldn’t. Since _self.app.event_producer_ is _MockEventProducer_, we can change the assertion like this:

```suggestion
                self.assertIsNone(self.app.event_producer.event)
```